backend/internal/service/audit_log_service.go

```diff
@@ -150,6 +150,14 @@ func (s *AuditLogService) ListAllAuditLogs(ctx context.Context, sortedPagination
 			return nil, utils.PaginationResponse{}, fmt.Errorf("unsupported database dialect: %s", dialect)
 		}
 	}
+	if filters.Location != "" {
+		switch filters.Location {
+		case "external":
+			query = query.Where("country != 'Internal Network' AND country IS NOT NULL AND country != ''")
```

Comment:
Can you please create an index on `country`? Right now, it's not indexed, so this query would always perform a full table scan